src/core/experimental/memutils.d

```diff
@@ -0,0 +1,352 @@
+    {
+        Dmemset_naive(d, cast(const(ubyte))val, n);
+    }
+}
+else
+{
+    version (D_SIMD)
+    {
+        // NOTE(stefanos): I could not GDC respective intrinsics.
+        void Dmemset(void *d, const uint val, size_t n)
+        {
+            import core.simd: int4;
+            version (LDC)
+            {
+                import ldc.simd: loadUnaligned, storeUnaligned;
+            }
+            else
+            version (DigitalMars)
+            {
+                import core.simd: void16, loadUnaligned, storeUnaligned;
+            }
+            else
+            {
+                static assert(0, "Only DMD / LDC are supported");
+            }
+            // TODO(stefanos): Is there a way to make them @safe?
+            // (The problem is that for LDC, they could take int* or float* pointers
+            // but the cast to void16 for DMD is necessary anyway).
+            void store32i_sse(void *dest, int4 reg)
+            {
+                version (LDC)
+                {
+                    storeUnaligned!int4(reg, cast(int*)dest);
+                    storeUnaligned!int4(reg, cast(int*)(dest+0x10));
+                }
+                else
+                {
+                    storeUnaligned(cast(void16*)dest, reg);
+                    storeUnaligned(cast(void16*)(dest+0x10), reg);
+                }
+            }
+            void store16i_sse(void *dest, int4 reg)
+            {
+                version (LDC)
+                {
+                    storeUnaligned!int4(reg, cast(int*)dest);
+                }
+                else
+                {
+                    storeUnaligned(cast(void16*)dest, reg);
+                }
+            }
+            // TODO(stefanos): Can we broadcast an int in a float4? That would be useful
+            // because then we would use only the float versions.
+            void broadcast_int(ref int4 xmm, int v)
+            {
+                xmm[0] = v;
+                xmm[1] = v;
+                xmm[2] = v;
+                xmm[3] = v;
+            }
+            const uint v = val * 0x01010101;            // Broadcast c to all 4 bytes
+            // NOTE(stefanos): I use the naive version, which in my benchmarks was slower
+            // than the previous classic switch. BUT. Using the switch had a significant
+            // drop in the rest of the sizes. It's not the branch that is responsible for the drop,
+            // but the fact that it's more difficult to optimize it as part of the rest of the code.
+            if (n <= 16)
+            {
+                Dmemset_naive(cast(ubyte*)d, cast(ubyte)val, n);
+                return;
+            }
+            void *temp = d + n - 0x10;                  // Used for the last 32 bytes
+            int4 xmm0;
+            // Broadcast v to all bytes.
+            broadcast_int(xmm0, v);
+            ubyte rem = cast(ulong)d & 15;              // Remainder from the previous 16-byte boundary.
```

Comment:
No need to extend to ulong for 32-bit platforms, cast to size_t instead.

Reply:
Yes, actually, that should be `cast(ubyte)` anyway.